server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -396,12 +404,51 @@ export const ExpensesCollectionQueryResolver = async (
   } else if (args.tag === null || args.tags === null) {
     where['tags'] = { [Op.is]: null };
   }
-  if (args.minAmount) {
-    where['amount'] = { [Op.gte]: args.minAmount };
-  }
-  if (args.maxAmount) {
-    where['amount'] = { ...where['amount'], [Op.lte]: args.maxAmount };
+
+  if (args.amount?.gte || args.amount?.lte) {
+    if (args.amount.gte && args.amount.lte) {
+      assert(args.amount.gte.currency === args.amount.lte.currency, 'Amount range must have the same currency');
```

Comment:
That's pure evil
> { gte: { value: 10, currency: 'USD' }, lte: { value: 100: currency: 'INR' } }

![giphy](https://github.com/user-attachments/assets/d7649b54-b751-4c32-9823-8546ca404eca)

Reply:
If someone forges a query like that, I'll consider it a DoS attempt and ban them from the platform.